tests/unit/helpers/events.test.js

```diff
@@ -240,19 +249,69 @@ describe('updateEvent', () => {
 
     await EventHelper.updateEvent(event, payload, credentials);
 
-    sinon.assert.called(updateRepetition);
+    sinon.assert.calledOnceWithExactly(updateRepetition, event, payload, credentials);
+    sinon.assert.notCalled(isRepetition);
     EventMock.verify();
   });
 
   it('should update event', async () => {
+    const companyId = new ObjectID();
+    const credentials = { _id: new ObjectID(), company: { _id: companyId } };
+    const eventId = new ObjectID();
+    const auxiliaryId = new ObjectID();
+    const event = { _id: eventId, type: INTERVENTION, auxiliary: { _id: auxiliaryId } };
+    const payload = {
+      startDate: '2019-01-21T09:38:18',
+      endDate: '2019-01-21T12:38:18',
+      auxiliary: auxiliaryId.toHexString(),
+      misc: 'test',
+    };
+
+    isUpdateAllowed.returns(true);
+    isMiscOnlyUpdated.returns(false);
+    isRepetition.returns(false);
+    formatEditionPayload.returns({ $set: {}, unset: {} });
+    EventMock.expects('updateOne').withExactArgs({ _id: event._id }, { $set: {}, unset: {} }).once();
+    EventMock.expects('findOne')
+      .withExactArgs({ _id: event._id })
+      .chain('populate')
+      .withExactArgs({
+        path: 'auxiliary',
+        select: 'identity administrative.driveFolder administrative.transportInvoice company picture',
+        populate: { path: 'sector', select: '_id sector', match: { company: companyId } },
+      })
+      .chain('populate')
+      .withExactArgs({ path: 'customer', select: 'identity subscriptions contact' })
+      .chain('populate')
+      .withExactArgs({ path: 'internalHour', match: { company: companyId } })
+      .chain('lean')
+      .once()
+      .returns({ ...event, updated: 1 });
+
+    await EventHelper.updateEvent(event, payload, credentials);
+
+    EventMock.verify();
+    sinon.assert.calledOnceWithExactly(populateEventSubscription, { ...event, updated: 1 });
+    sinon.assert.calledOnceWithExactly(formatEditionPayload, event, payload, false);
+    sinon.assert.calledOnceWithExactly(isUpdateAllowed, event, payload, credentials);
+    sinon.assert.calledOnceWithExactly(isRepetition, event);
+    sinon.assert.calledOnceWithExactly(isMiscOnlyUpdated, event, payload);
+    sinon.assert.notCalled(deleteConflictInternalHoursAndUnavailabilities);
+    sinon.assert.notCalled(unassignConflictInterventions);
+    sinon.assert.notCalled(updateRepetition);
+  });
+
+  it('should update event when misc is updates among other fields', async () => {
```

Comment:
updated

Reply:
en dessous aussi

Reply:
tu pourrais ajouter "and event is a repetition" pour comparer avec celui d'en dessous